File: Front-End/src/app/models/FormaPagamento.ts
export enum FormaPagamento {
    CREDITO = "Crédito",
    DEBITO = "Débito",
    PIX = "Pix",
    BOLETO = "Boleto",
    DINHEIRO = "Dinheiro"
}

export function getDescricaoFormaPagamento(forma: FormaPagamento): string {
    return forma;
}

export function getFormaPagamentoPorDescricao(descricao: string ): FormaPagamento | null {
    if (descricao === null) return null;

    for (const [key, value] of Object.entries(FormaPagamento)) {
        if (value === descricao) {
            return key as FormaPagamento;
        }
    }
    return null; 
}

File: Back-End/src/main/java/com/olhardigital/backend/Model/Utils/FormaPagamento.java
package com.olhardigital.backend.Model.Utils;

public enum FormaPagamento {
    CREDITO("Crédito"),
    DEBITO("Débito"),
    PIX("Pix"),
    BOLETO("Boleto"),
    DINHEIRO("Dinheiro");

    private final String descricao;

    FormaPagamento(String descricao) {
        this.descricao = descricao;
    }

    public String getDescricao() {
        return descricao;
    }

    @Override
    public String toString() {
        return descricao;
    }
}
